edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Diagnostics/MetricFilter.cs

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.Diagnostics
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Newtonsoft.Json;
+
+    /// <summary>
+    /// This class acts as a pass through for a group of metrics.
+    /// It will exclude metrics based on a tag whitelist and blacklist.
+    /// After filtering, it will add or remove tags.
+    /// </summary>
+    public class MetricFilter
+    {
+        Option<List<KeyValuePair<string, string>>> tagsWhitelist = Option.None<List<KeyValuePair<string, string>>>();
```

Comment:
IIRC, Whitelist and blacklist are restricted keywords and will flag SDL scans. Can you change those to "allowed" and "blocked" or something simliar to that?

Reply:
Do we run SDL scans in check-in? It's been passing the pr build. If possible, I'd like to leave it whitelist because I think it's more clear.

Reply:
Despite the inconvenience, I think it's better to find a different name. `allowedTags`, `filteredTags`, `tagsList`, `tagsFilter`, ...